lxd/auth/rbac/rbac.go

```diff
@@ -74,8 +71,8 @@ type Server struct {
 }
 
 // NewServer returns a new RBAC server instance.
-func NewServer(apiURL string, apiKey string, agentAuthURL string, agentUsername string, agentPrivateKey string, agentPublicKey string) (*Server, error) {
-	r := Server{
+func NewServer(apiURL string, apiKey string, agentAuthURL string, agentUsername string, agentPrivateKey string, agentPublicKey string, getProjects func() (map[int64]string, error)) (auth.Authorizer, error) {
```

Comment:
I would prefer to see this function change and the internal changes as separate commits so that its easier to see the implementation change and then the update usage of that implementation as the next commit, otherwise as the reviewer you only see the implementation change buried half way down in the mechanical changes.

Reply:
I've created two separate commits, one for the rbac change, and one for the Authorizer usage across the code base.